micro-rdk/src/common/conn/viam.rs

```diff
@@ -734,35 +774,58 @@ where
         mut app_client: Option<AppClient>,
     ) -> Result<(), errors::ServerError> {
         let wait = Duration::from_secs(1); // should do exponential back off
+
         loop {
-            if let Some(app_client) = app_client {
-                let mut app_client_tasks: FuturesUnordered<AppClientTaskRunner> =
-                    FuturesUnordered::new();
+            if let Some(app_client) = &app_client {
+                log::info!("starting execution of app client tasks");
+                let mut app_client_tasks: FuturesUnordered<_> = FuturesUnordered::new();
                 for task in &self.app_client_tasks {
-                    app_client_tasks.push(AppClientTaskRunner {
-                        app_client: &app_client,
-                        invoker: task,
-                        state: TaskRunnerState::Run {
-                            task: task.invoke(&app_client),
-                        },
+                    app_client_tasks.push({
+                        async move {
+                            let mut duration = task.get_default_period();
+                            loop {
+                                let _ = Timer::after(duration).await;
+                                match task.invoke(app_client).await {
+                                    Ok(Some(next_duration)) => {
+                                        duration = next_duration;
+                                    }
+                                    Ok(None) => {}
+                                    e @ Err(_) => {
+                                        break e;
+                                    }
+                                }
+                            }
+                        }
                     });
```

Comment:
noting here from offline discussion,

- `self.app_client_tasks` are set early in startup and don't change
- below where we iterate through the `app_client_tasks`, any error from app_client is an issue for all futures in the Vec. 
  -  (maybe should rename to `task_queue` or something to distinguish from `self.app_client_tasks`)
- we then attempt to create a new app client connection and insert it into the `mut app_client` parameter that was passed in.